src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -8,16 +9,21 @@
     /// </summary>
     public abstract class BehaviorContext : ContextBag
     {
+        //TODO DanielTim: Does it make to force us to provide a RootContext here (to ensure Settings are set)? RootContext currently is private.
         /// <summary>
         /// Create an instance of <see cref="BehaviorContext"/>.
         /// </summary>
         /// <param name="parentContext">The parent context.</param>
-        protected BehaviorContext(ContextBag parentContext) : base(parentContext)
+        protected BehaviorContext(BehaviorContext parentContext) : base(parentContext)
         {
-          
         }
 
         /// <summary>
+        /// The readonly settings.
+        /// </summary>
+        public ReadOnlySettings Settings => Get<ReadOnlySettings>();
```

Comment:
That's fine, but can't that be a separate "Dictionary<string,object>" ? 

This was we force us/users to read the settings in the `Feature.Setup` and translate that to runtime concepts?